src/algorithm/verifySecureUriClaim.ts

```diff
@@ -9,9 +9,10 @@ import { SecureUriClaimVerificationError } from "../error/SecureUriClaimVerifica
  * See also: https://solid.github.io/solid-oidc/#security-tls & https://github.com/solid/authentication-panel/issues/114#issuecomment-751867437
  */
 export function verifySecureUriClaim(uri: string, claim: string): void {
+  const url = new URL(uri);
   if (
-    !uri.startsWith("https://") &&
-    !/^http:\/\/([\w-]+.)*localhost:.*/.test(uri)
+    url.protocol !== "https:" &&
+    url.hostname.split(".").pop() !== "localhost"
```

Comment:
@RubenVerborgh What do you think about something like this?

Reply:
That seems indeed more rigorous 👍 (Might still want to check for `http` in the second case; all kinds of attacks exist.)

Reply:
I think @Limpan's [latest commit](https://github.com/CommunitySolidServer/access-token-verifier/pull/236/commits/4af407559973d4699b97e40cff888077ff13c85e) addresses all concerns.